plugins/scaffolder-actions/scaffolder-backend-module-http-request/src/actions/run/HttpClient.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Logger} from "winston";
+import {fetch} from "cross-fetch";
+
+class HttpError extends Error {}
+const DEFAULT_TIMEOUT = 60_000;
+
+export type HttpBackstageRequestAuthorizer = (request: RequestInit) => RequestInit
+
+export class HttpClient {
+    private readonly logger: Logger;
+    private readonly authorizer: HttpBackstageRequestAuthorizer;
+    constructor({ logger, authorizer }: { logger: Logger, authorizer: HttpBackstageRequestAuthorizer }) {
+        this.logger = logger;
+        this.authorizer = authorizer;
+    }
+
+    async request(url: string, options: RequestInit) {
+        let res: any;
+        const controller = new AbortController()
+        const timeoutId = setTimeout(() => controller.abort(), DEFAULT_TIMEOUT)
+        const httpOptions = {...options, signal: controller.signal}
+
+        try {
+            res = await fetch(url, this.authorizer(httpOptions));
```

Comment:
`this.authorizer(httpOptions)` is the thing that actually adds the authorization header to the request before sending it. It can be injected in.